### ISSUE TITLE
Update documentation for npc/actions

### ISSUE THREAD

**Issue by junasano** [uchicago-cs/chiventure#1573]
The current documentation was created for an older version of npc actions, therefore, the wiki page documentation needs to be updated accordingly for the next groups.

**Comment by junasano** [comment]
The first PR was approved and merged and the second (and final) PR was approved by a peer and is waiting on the TA's approval. 
Since it is close to being fully implemented and I will go forward with editing the wiki page.

**Comment by junasano** [comment]
The overall wiki update is now complete at: 
https://github.com/uchicago-cs/chiventure/wiki/NPC-~-Action-Documentation
However, links to the backlog has to be still created as well as making sure this documentation shows up on the side bar.

**Comment by junasano** [comment]
The two cards in the back log have now been linked to the wiki page. And the wiki page also shows up on the side bar. 
The main part I changed was the structure section since the agent_t was not mentioned before and the linked list is also new. A whole new section of future work was also added.
Furthermore, the introduction was slightly modified to reflect the new names of the functions an the implementation part was updated to account for the new functions.

**Comment by lilyehsani** [comment]
Issue Score: Excellent

Comments:
Great work on this issue! The summary for this issue is too terse. A good rule of thumb is that you should be writing at least one full paragraph in your summary (see https://uchicago-cs.github.io/cmsc22000/project/issue-pr-grading.html#descriptive-summary for examples).
